IO/NativeSignalHandlers/WindowsNTHandler.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Runtime.InteropServices;
+
+namespace LocalAdmin.V2.IO.NativeSignalHandlers
+{
+    /// <summary>
+    ///     Native signal processing on Windows NT systems.
+    /// </summary>
+    internal sealed class WindowsNTHandler : INativeSignalHandler
+    {
+        public static readonly WindowsNTHandler Handler;
+
+        static WindowsNTHandler()
+        {
+            Handler = new WindowsNTHandler();
+        }
+
+        private WindowsNTHandler() { }
+
+        public void Setup()
+        {
+            SetConsoleCtrlHandler(OnNativeSignal, true);
+        }
+
+        private bool OnNativeSignal(CtrlTypes ctrl)
+        {
+            Core.LocalAdmin.Singleton.Exit(0);
+            return true;
+        }
+
+        #region Native
+
+        [DllImport("Kernel32")]
+        private static extern bool SetConsoleCtrlHandler(HandlerRoutine handler, bool add);
```

Comment:
You should handle errors here

Reply:
I do not think it will be useful, the main exception handling is in the method `Start` of the class `LocalAdmin`, call stack equals
```js
INativeSignalHandler::Setup()
LocalAdmin::SetupPlatform
LocalAdmin::Start
Program::Start
```
if the current operating system is Windows NT, then `WindowsNTHandler :: Setup` is called, also `LocalAdmin::SetupPlatform` is called immediately after receiving the port to start the game, this means that if it throws an exception, the game will not be launched, you propose to handle the exception here, but what will happen next, will you allow the game to start without processing the exit from the console? The game will just hang in the background without a console and in most cases no one will know about it, and the next time start the game there will be an error trying to forward the port. I don't approve.

Reply:
At least print the error out when it happens, now the users have no idea if it fails

Reply:
Yes it already
```csharp
            catch (Exception ex)
            {
                File.WriteAllText($"{DateTimeOffset.UtcNow:yyyy-MM-ddTHH-mm-ssZ}-crash.txt", ex.ToString());

                /*
                Logger.Log("|===| Exception |===|");
                Logger.Log("Time: " + DateTime.Now);
                Logger.Log(ex);
                Logger.Log("|===================|");
                Logger.Log("");
                */
            }
```

Reply:
No, cause you don't handle the case when this method returns false, which means that it failed

Reply:
Woah, I'll fix it